### PR TITLE
Viewport

### PR DESCRIPTION
**Pull request by evil159**
### What does this pull request do?

This pull request introduces viewport support to the MapWidget, allowing developers to control the camera’s initial position and behavior when displaying a map. By specifying a ViewportState subclass in the viewport parameter, developers can center the map on specific locations, follow the user’s position, or display an overview of a geometry.
```dart
MapWidget(
  viewport: CameraViewportState(
    center: Point(coordinates: Position(-117.918976, 33.812092)),
    zoom: 15.0,
  ),
);
```

### What is the motivation and context behind this change?

https://mapbox.atlassian.net/browse/MAPSFLT-60

### Pull request checklist:
 - [x] Add a changelog entry.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [x] Add documentation comments for any added or updated public APIs.

### CODE REVIEW

File: lib/src/viewport/states/camera_viewport_state.dart [hunk 1]
```diff
@@ -0,0 +1,40 @@
+part of mapbox_maps_flutter;
+
+/// Manually sets camera to specified properties.
+final class CameraViewportState extends ViewportState {
```
Comment (pjleonard37) [inline lib/src/viewport/states/camera_viewport_state.dart:4]: I get a crash anytime I try to use this (like in the simple example from the Changelog below). 

> Exception has occurred.
> ArgumentError (Invalid argument: Instance of 'CameraOptions')

```
MapWidget(
  viewport: CameraViewportState(
    center: Point(coordinates: Position(-117.918976, 33.812092)),
    zoom: 15.0,
  ),
);
```

Comment (evil159) [reply lib/src/viewport/states/camera_viewport_state.dart:4]: Whoops, thanks for catching this! I must have forgotten to push forward declarations for types in CameraViewportState, fixed!

File: lib/src/map_widget.dart [hunk 2]
```diff
@@ -69,9 +69,8 @@ class MapWidget extends StatefulWidget {
     this.onTapListener,
     this.onLongTapListener,
     this.onScrollListener,
-  }) : super(key: key) {
```
[the remark this answers is not in the document]
Comment (evil159) [reply lib/src/map_widget.dart:72]: Moved to `initState()`

File: lib/src/viewport/transitions/fly_viewport_transition.dart [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+/// );
+/// ```
+///
+/// In this example, the fly animation will occur over 5 seconds. If [duration] is not specified,
+/// the transition will use a default duration determined by the system.
+final class FlyViewportTransition extends ViewportTransition {
+  /// The duration of the fly animation.
+  ///
+  /// Specifies how long the fly animation should take to complete. If not provided, a default duration
+  /// will be used by the system.
```
Comment (pjleonard37) [inline lib/src/viewport/transitions/fly_viewport_transition.dart:23]: Which system here? Is this platform specific?

File: example/lib/simple_map_example.dart [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  State<StatefulWidget> createState() => _SimpleMapState();
+}
+
+class _SimpleMapState extends State<SimpleMapExample> {
+  _SimpleMapState();
+
+  @override
+  Widget build(BuildContext context) {
+    return MapWidget(
+      styleUri: MapboxStyles.STANDARD,
```
Comment (pjleonard37) [inline example/lib/simple_map_example.dart:26]: Could we add a simple viewport here? Maybe: 

```
  viewport: CameraViewportState(
    center: Point(coordinates: Position(-117.918976, 33.812092)),
    zoom: 15.0,
  ),
```

File: example/lib/viewport_example.dart [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+              _cityIndex++;
+              _flying++;
+            },
+            transition: FlyViewportTransition(
+              duration:
+                  cities[(_cityIndex + 1) % cities.length].animationDuration,
+            ),
+            completion: (result) {
+              print(
+                  'Animaton complete with $result, currentCity ${currentCity.name}');
```
Comment (pjleonard37) [inline example/lib/viewport_example.dart:48]:
```suggestion
                  'Animation complete with $result, currentCity ${currentCity.name}');
```

File: lib/src/viewport/transitions/fly_viewport_transition.dart (hunk 3 above)
Comment (evil159) [reply lib/src/viewport/transitions/fly_viewport_transition.dart:23]: If not specified the duration is calculated by the fly to interpolation algorithm, should be the same on both platform as the algorithm is the same.

File: example/lib/simple_map_example.dart (hunk 4 above)
Comment (evil159) [reply example/lib/simple_map_example.dart:26]: Nice choice for the location 👍 Added!

File: example/integration_test/viewport_test.dart [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+      bearing: 8,
+      pitch: 42,
+    ));
+
+    // apply default viewport
+    app.main(viewport: StyleDefaultViewportState());
+    await tester.pumpAndSettle();
+    final cameraAfterViewport = await mapboxMap.getCameraState();
+
+    // check that the came is reset to the default viewport
```
Comment (pjleonard37) [inline example/integration_test/viewport_test.dart:64]:
```suggestion
    // check that the camera is reset to the default viewport
```

File: example/integration_test/viewport_test.dart [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+    final followPuckViewport = FollowPuckViewportState(
+      zoom: 16.0,
+      bearing: FollowPuckViewportStateBearingConstant(30.0),
+      pitch: 45.0,
+    );
+    final mapFuture = app.main(viewport: followPuckViewport);
+    await tester.pumpAndSettle();
+    final _ = await mapFuture;
+
+    // TODO: test that the camera is updated when custom location provider is supported
```
Comment (pjleonard37) [inline example/integration_test/viewport_test.dart:131]: Create a ticket for this?

**Review by pjleonard37 — APPROVED** [review]
Looks good! Excited for this :)

File: example/integration_test/viewport_test.dart (hunk 7 above)
Comment (evil159) [reply example/integration_test/viewport_test.dart:131]: The ticket is [here](https://mapbox.atlassian.net/browse/MAPSFLT-274)

**Review by pjleonard37 — APPROVED** [review]

**Pull request merged**